node/src/effect.rs

```diff
@@ -874,28 +878,20 @@ impl<REv> EffectBuilder<REv> {
         .await
     }
 
-    /// Announces that a proto block has been proposed and will either be finalized or orphaned
-    /// soon.
-    pub(crate) async fn announce_proposed_proto_block(self, proto_block: ProtoBlock)
-    where
-        REv: From<ConsensusAnnouncement>,
-    {
-        self.0
-            .schedule(
-                ConsensusAnnouncement::Proposed(proto_block),
-                QueueKind::Regular,
-            )
-            .await
-    }
-
     /// Announces that a proto block has been finalized.
-    pub(crate) async fn announce_finalized_block(self, finalized_block: FinalizedBlock)
-    where
+    pub(crate) async fn announce_finalized_block(
+        self,
+        finalized_block: FinalizedBlock,
+        parent: Option<ProtoBlockHash>,
```

Comment:
Should we maybe be using `FinalizedBlock` hashes instead of `ProtoBlockHash` everywhere? That would be unique, would it?

Reply:
Should be. Or maybe block height (see above)?